### PR TITLE
Fix docs: regarding max_clause_count

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Doc fix - setting `max_clause_count` to the size of the array does not guarantee any operations on it will not trigger `TooManyClauses`.

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/LuceneQueryBuilderIntegrationTest.java
Comment (jeeminso) [inline server/src/test/java/io/crate/integrationtests/LuceneQueryBuilderIntegrationTest.java:357]: Without the padding, these two fail with `TooManyClauses`

**Review by mfussenegger — COMMENTED** [review]
I'm not sure about this. The `max_clause_count` setting is a global setting affecting all kinds of queries. I don't think adding a padding (that's just one?) is right.
If we have cases where we create a `BooleanQuery` made up out of other queries which themselves are made up out of `n` queries, then the `n` would act as a multiplier.

I'm not sure what to do about this. Maybe we just document this in a bit more detail with an example?

@seut @matriv what do you think?

**Comment by matriv** [comment]
I'm also not sure, in the documentation: 
https://github.com/crate/crate/blob/master/docs/general/dql/selects.rst?plain=1#L658 https://github.com/crate/crate/blob/master/docs/general/dql/selects.rst?plain=1#L688 we describe the issue, but as @mfussenegger pointed out it's not accurate for more complex expressions
```
cr> select count(*) from t2 where id1 != any(select * from generate_series(1, 8000, 1)) and id1 != any(select * from generate_series(1, 191, 1));
+----------+
| count(*) |
+----------+
|        1 |
+----------+
SELECT 1 row in set (0.015 sec)
cr> select count(*) from t2 where id1 != any(select * from generate_series(1, 8000, 1)) and id1 != any(select * from generate_series(1, 192, 1));
TooManyNestedClauses[Query contains too many nested clauses; maxClauseCount is set to 8192]
```

So I think it's better to just update the documentation and not mention that `8192` directly refers to maximum element in the array, but just say that the size of the array is limited and depends on the setting.

**Comment by seut** [comment]
> So I think it's better to just update the documentation and not mention that 8192 directly refers to maximum element in the array, but just say that the size of the array is limited and depends on the setting.

I also think this is the best solution.

**Comment by jeeminso** [comment]
Thank you all for your inputs, please check the updated docs.

File: docs/config/node.rst
Comment (matriv) [inline docs/config/node.rst:857]:
```suggestion
  prevent users from executing queries that may result in heavy memory consumption
```

File: docs/config/node.rst
Comment (matriv) [inline docs/config/node.rst:859]:
```suggestion
  ``TooManyClauses`` errors when the limit is exceeded.
```
From a quick look we don't use `exceptions` in docs as this is Java specific and we prefer the more generic: `errors`.

File: docs/config/node.rst
Comment (matriv) [inline docs/config/node.rst:864]: I wouldn't mention anything about a formula, I think something like this:
`the number of boolean clauses used can be larger than the elements of the array`
is enough.

File: docs/general/dql/selects.rst
Comment (matriv) [inline docs/general/dql/selects.rst:658]:
```suggestion
    When ``!= ANY(<array_col>))`` causes ``TooManyClauses`` error you
```

File: docs/general/dql/selects.rst
Comment (matriv) [inline docs/general/dql/selects.rst:659]:
```suggestion
    should consider increasing :ref:`indices.query.bool.max_clause_count setting.
```

File: docs/general/dql/selects.rst
Comment (matriv) [inline docs/general/dql/selects.rst:688]:
```suggestion
    arrays causes ``TooManyClauses`` errors, you must configure
```

**Review by matriv — COMMENTED** [review]
Thx @jeeminso, left some comments, and also please wait for one more review by @mfussenegger

File: docs/config/node.rst
Comment (matriv) [inline docs/config/node.rst:863]: I think it reads better:
```suggestion
    You can avoid ``TooManyClauses`` errors by increasing this setting. The number of
```

File: docs/general/dql/selects.rst
Comment (matriv) [inline docs/general/dql/selects.rst:688]:
```suggestion
    arrays causes ``TooManyClauses`` errors, you could consider increasing
```
I think `could` is more accurate, as we should not encourage users to increase the limit, but rather consider rewriting their queries or even have a better schema where those nested `IN()` subqueries are not used.

**Review by matriv — APPROVED** [review]

File: docs/general/dql/selects.rst
Comment (seut) [inline docs/general/dql/selects.rst:658]: same here (what @matriv mentioned below)
```suggestion
    When ``!= ANY(<array_col>))`` causes ``TooManyClauses`` errors you could
```

**Review by seut — APPROVED** [review]
Lgtm otherwise

**Pull request merged**
